src/Filesystem/Folder.php

```diff
@@ -631,20 +631,18 @@ public function create($pathname, $mode = false)
         $pathname = rtrim($pathname, DIRECTORY_SEPARATOR);
         $nextPathname = substr($pathname, 0, strrpos($pathname, DIRECTORY_SEPARATOR));
 
-        if ($this->create($nextPathname, $mode)) {
-            if (!file_exists($pathname)) {
-                $old = umask(0);
-                if (mkdir($pathname, $mode, true)) {
-                    umask($old);
-                    $this->_messages[] = sprintf('%s created', $pathname);
-
-                    return true;
-                }
+        if ($this->create($nextPathname, $mode) && !file_exists($pathname)) {
+            $old = umask(0);
+            if (mkdir($pathname, $mode, true)) {
                 umask($old);
-                $this->_errors[] = sprintf('%s NOT created', $pathname);
+                $this->_messages[] = sprintf('%s created', $pathname);
 
-                return false;
+                return true;
             }
+            umask($old);
+            $this->_errors[] = sprintf('%s NOT created', $pathname);
+
+            return false;
```

Comment:
Existing bail early is better practice.
Also such code changes unless accompanied by a specific unit test can introduce silent bugs